lib/beacon-verification.ts

```diff
@@ -66,13 +71,14 @@ export async function verifySigOnG1(
     domainSeparationTag= 'BLS_SIG_BLS12381G2_XMD:SHA-256_SSWU_RO_NUL_'
 ): Promise<boolean> {
     const P = normP2(publicKey);
-    const Hm = await normP1Hash(message, domainSeparationTag);
-    const G = PointG2.BASE;
+    const Hm = normP1Hash(message, domainSeparationTag);
+    const G = bls.G2.ProjectivePoint.BASE;
     const S = normP1(signature);
-    const ePHm = pairing(Hm, P.negate(), false);
-    const eGS = pairing(S, G, false);
-    const exp = eGS.multiply(ePHm).finalExponentiate();
-    return exp.equals(Fp12.ONE);
+    const ePHm = bls.pairing(Hm, P.negate(), false);
+    const eGS = bls.pairing(S, G, false);
+    const Fp12 = bls.fields.Fp12 as any;// as {finalExponentiate(num: any): any;};
+    const exp = Fp12.finalExponentiate(bls.fields.Fp12.mul(eGS, ePHm));
```

Comment:
`finalExponentiate` is necessary. Tests pass, but we need to figure out a type-safe way to access it.

Reply:
I've raised an issue on noble-curves: https://github.com/paulmillr/noble-curves/issues/65

If the build passes, I think we can merge this for now (after you remove the commented out bit) and I'll raise an issue to fix it if Paul gives us another route to do so or exposes it for us.